packages/frontend/src/features/virtualView/components/HeaderVirtualView.tsx

```diff
@@ -337,8 +338,17 @@ export const HeaderVirtualView: FC<{
     const history = useHistory();
     const sql = useAppSelector((state) => state.sqlRunner.sql);
     const projectUuid = useAppSelector((state) => state.sqlRunner.projectUuid);
-    const hasUnrunChanges = useAppSelector(
-        (state) => state.sqlRunner.hasUnrunChanges,
```

Comment:
why can't you rely on `state.sqlRunner.hasUnrunChanges` ? is this not duplicating the logic ?

Reply:
can't rely on this because this was the wrong selector for this behaviour. 
This just checks if there are unrunChanges, not if there's a change in original sql vs current sql. 
A similar behaviour is done on the `save` behaviour in `HeaderEdit` 
```tsx
disabled={!config || !sql || !hasChanges}
```
where `hasChanges` is calculated on the component: 

```tsx

    const hasChanges = useMemo(() => {
        if (!initialSavedSqlChart) return false;
        const changedSql = sql !== initialSavedSqlChart.sql;
        const changedLimit = limit !== initialSavedSqlChart.limit;
        const changedConfig = !isEqual(config, initialChartConfig);

        return changedSql || changedLimit || changedConfig;
    }, [initialSavedSqlChart, initialChartConfig, sql, limit, config]);
```

you can see here that we also check the sql against the initial sql